scripts/platform-init/platform-init.py

```diff
@@ -109,10 +129,22 @@ def init_secret_service_secret(config: Config):
             ),
         )
     else:
-        # just load key to create public key from
+        # just load key to create public key from and unset previous secret if set
+        logging.info("Leaving private key unchanged")
         private_key = serialization.load_pem_private_key(
             existing_private_key.encode(), password=None
         )
+        v1.patch_namespaced_secret(
```

Comment:
question: My understanding is that in this branch of the logic nothing has changed, so why do we have to patch the secret? Isn't  `existing_private_key` just the result of reading the `private_key_secret`? So this patches the contents of the secret with the same thing that is already there.

Reply:
if we rotated secrets before, there's an old and new key in the secret, and this removes the old key entry (otherwise secrets storage would just try to rotate again, which would be a nop but still take cpu cycles/hit the DB)

Reply:
Another solution would be to use two separate k8s secrets for the new and old key and delete the old one. but that's one of the annoying parts with the platform init script, since it happens after helm deploy, in the chart it's harder to conditionally mount secrets as at deploy we don't know if there will be a secret. We can read the config values from the values.yaml and then parse them and do some {{ if ... }} based on them, but then we have python code and helm template that have logic based on a value which isn't that clean. Just having one secret with variable properties that's always mounted is easier and then it's just handled in one place.